### PR TITLE
My recent swaps v2 doc

### PR DESCRIPTION
**Pull request by harsenyan3**
Documentation for v2 my recent swaps rpc method. Issue #280 .

### CODE REVIEW

**Comment by gcharang** [comment]
I think the file name doesn't need v2 as the url already has it

File: src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+**my\_recent\_swaps (from\_uuid page\_number=1 limit=10 my\_coin other\_coin from\_timestamp to\_timestamp)**
+
+The `my_recent_swaps` method returns the data of the most recent atomic swaps executed by the Komodo DeFi Framework API node. Please note that all filters (my\_coin, from\_timestamp, etc.) are combined using logical AND.
+
+## Arguments
+
+| Structure       | Type   | Description                                                                                                                            |
+| --------------- | ------ | -------------------------------------------------------------------------------------------------------------------------------------- |
+| filter          | Object | [FilterCriteria](/komodo-defi-framework/api/common_structures/#filter-criteria) containing all the filtering criteria for the swaps    |
+| paging\_options | Object | [PagingOptionsObject](/komodo-defi-framework/api/common_structures/#paging-options) containing all the paging options for the request. |
```
Comment (shamardy) [inline src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx:15]: filter and paging_options are flattened structs ref. https://github.com/KomodoPlatform/komodo-defi-framework/blob/af571608c34f38d82e3c421f9bd61c5265db0cf8/mm2src/mm2_main/src/lp_swap/swap_v2_rpcs.rs#L373-L379
Which mean the json should look as something like this
```json
{
  "my_coin": "BTC",
  "other_coin": "ETH",
  "from_timestamp": 1622505600,
  "to_timestamp": 1625097600,
  "limit": 10,
  "page_number": 1,
  "from_uuid": "550e8400-e29b-41d4-a716-446655440000"
}
```

File: src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+  ```json
+  {
+    "mmrpc": "2.0",
+    "userpass": "RPC_UserP@SSW0RD",
+    "method": "my_recent_swaps",
+    "params": {
+      "filter": {
+        "status": "completed",
+        "date_from": "2024-01-01T00:00:00Z",
+        "date_to": "2024-07-01T00:00:00Z",
```
Comment (shamardy) [inline src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx:44]: These fields are not in `my_recent_swaps` v2 request

File: src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+  ```
+
+  Response (error)
+
+  ```json
+  {
+    "error": "lp_swap:1454] from_uuid e299c6ece7a7ddc42444eda64d46b163eaa992da65ce6de24eb812d715184e41 swap is not found"
+  }
+  ```
+</CollapsibleSection>
```
Comment (shamardy) [inline src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx:390]: I believe this error should be `from_uuid e299c6ece7a7ddc42444eda64d46b163eaa992da65ce6de24eb812d715184e41 is not found` , the word `swap` is not in the error.

**Review by shamardy — COMMENTED** [review]
Thanks for the PR! First review iteration!

File: src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx (hunk 1 above)
Comment (smk762) [reply src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx:15]: Is this consistent across all methods? I see `PagingOptionsEnum` is within existing codebase for lightning and some tx_history methods

File: src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx (hunk 3 above)
Comment (smk762) [reply src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx:385]: I am unable to replicate this error. Currently, using `"from_uuid": "e299c6ece7a7ddc42444eda64d46b163eaa992da65ce6de24eb812d715184e40"` (which does not exist) still returns other recent swaps.

File: src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx (hunk 1 above)
Comment (shamardy) [reply src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx]: > Is this consistent across all methods?

No, but it should be, I understand your point. For lightning and tx_history it's not flattened. BTW, this flattened structs are the same structs that are used for `my_recent_swaps` legacy/v1 so it's exactly the same request as v1.

File: src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx (hunk 3 above)
Comment (shamardy) [reply src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx:384]: Right, it doesn't return an error, it ignores the uuid as if it's not there and just returns the results from the start or row 0. This should be the case for legacy/v1 RPC as well, please check this as this error is in one of the responses in the v1 docs as well.

Comment (smk762) [reply src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx]: I've removed that error from the v2 doc, and updated the v1 doc with a couple of current error examples. Using a non-existent uuid (in the correct format) returns

```json
{
    "error": "rpc:215] dispatcher_legacy:141] lp_swap:1299] my_swaps_storage:102] Unknown SQL error: Query returned no rows"
}
```

File: src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+**my\_recent\_swaps (from\_uuid page\_number=1 limit=10 my\_coin other\_coin from\_timestamp to\_timestamp)**
+
+The `my_recent_swaps` method returns the data of the most recent atomic swaps executed by the Komodo DeFi Framework API node. Please note that all filters (my\_coin, from\_timestamp, etc.) are combined using logical AND.
+
+## Arguments
+
+| Structure | Type   | Description                                                                                                                         |
+| --------- | ------ | ----------------------------------------------------------------------------------------------------------------------------------- |
+| filter    | Object | [FilterCriteria](/komodo-defi-framework/api/common_structures/#filter-criteria) containing all the filtering criteria for the swaps |
```
Comment (shamardy) [inline src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx:14]: `paging_options` is not included in the arguments. It should be flattened as discussed same for `filter`, it should be flattened.  See next comment.

File: src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+      "filter": {
+        "my_coin": "BTC",
+        "other_coin": "ETH",
+        "from_timestamp": 1672531200,
+        "to_timestamp": 1704067200,
+        "from_uuid": null,
+        "limit": 5,
+        "page_number": 1
+      }
+    }
```
Comment (shamardy) [inline src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx:49]: This is how it should be with all structs flattened
```json
    "params": {
        "my_coin": "BTC",
        "other_coin": "ETH",
        "from_timestamp": 1672531200,
        "to_timestamp": 1704067200,
        "from_uuid": null,
        "limit": 5,
        "page_number": 1
    }
```

File: src/pages/komodo-defi-framework/api/common_structures/index.mdx [hunk 6]
```diff
@@ -395,3 +395,57 @@ The `WithdrawFee` object varies depending on the coin or token type. Refer to th
...
+      "other_coin": "ETH",
+      "from_timestamp": 1672531200,
+      "to_timestamp": 1704067200
+    }
+  }
+  ```
+</CollapsibleSection>
+
+<Note>
+  In some cases (e.g. `my_recent_swaps`), parameters from the #PagingOptions object are included directly within the #FilterCriteria object.
```
Comment (shamardy) [inline src/pages/komodo-defi-framework/api/common_structures/index.mdx:426]: Please update this note.

**Review by shamardy — COMMENTED** [review]
A few more comments

File: src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx (hunk 4 above)
Comment (smk762) [reply src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx]: fixed

File: src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx (hunk 5 above)
Comment (smk762) [reply src/pages/komodo-defi-framework/api/v20/my_recent_swaps/index.mdx]: fixed

**Review by shamardy — APPROVED** [review]

**Review by gcharang — APPROVED** [review]

**Pull request merged**